### ISSUE TITLE
Xamarin.iOS Build Task Doesn't work

### ISSUE THREAD

**Issue by Garyljackson** [Microsoft/vso-agent-tasks#346]
Hi

The Xamarin.iOS task always returns an error: "Return code: 1"

I have done the following to try out the build task.

* Created a VSO project, and uploaded the code for the TaskyiOS_Starter sample app from Xamarin.
* Set up a build with the Xamarin.iOS as the first build step, and configured it to build for the simulator
* Installed Xamarin Studio on the Mac, and logged in with a valid trial account
* Installed the cross platform build agent on a mac, and connected it to VSO (I have this running in interactive mode)

When I queue up a new build, I can see it hit my agent, but it returns an error, and the log details the following:

```
-07-15T00:18:50.059Z: [command]/usr/bin/nuget restore /Users/Gary.Jackson/Projects/myagent/_work/build/0fbee2c4e562ab4b6d6b2db37f92b44ffe234542132e10f0e14be3d9fb4b37e0/repo/TaskyiOS_Starter.sln
 
2015-07-15T00:18:50.359Z: [command]'/Applications/Xamarin Studio.app/Contents/MacOS/mdtool' --verbose build --configuration:'Release|iPhoneSimulator' /Users/Gary.Jackson/Projects/myagent/_work/build/0fbee2c4e562ab4b6d6b2db37f92b44ffe234542132e10f0e14be3d9fb4b37e0/repo/TaskyiOS_Starter.sln
 
##[Error] 2015-07-15T00:18:50.919Z: Return code: 1
```

Removing the following lines from the constructor of the local "**toolrunner.js**" allows the task to run, so I'm not sure if this is a task bug, or possibly something with the toolrunner?

```javascript

if (toolPath.indexOf(' ') > 0) {
toolPath = '\'' + toolPath + '\'';
}

```

**Comment by chrisrpatterson** [comment]
Yang?

**Comment by bryanmacfarlane** [comment]
I'll investigate on the toolrunner line - that seems like an odd thing to do (if working with path module shouldn't need to manually append paths).  I'll find out the background on that block of code.  Yang, chat with me.

**Comment by bryanmacfarlane** [comment]
hmmm, looked again and it's just quoting ... we're investigating.  Thanks for finding the issue!

**Comment by bryanmacfarlane** [comment]
I think I know the issue ... toolrunner calls child.spawn which takes tool path arg and that api should handle.  No need to quote.  *But* we used to call child.exec (which had all the limits with output etc...) which constructed one long line so we needed that block.

I'll push a fix to vso-task-lib today.

**Comment by Garyljackson** [comment]
Awesome, thank you, it was a bit of a head scratcher for me as well, that line seemed to make sense..

**Comment by bryanmacfarlane** [comment]
https://github.com/Microsoft/vso-task-lib/commit/1bf38c685982a121062e83e9ce70d15ba2f411e5

**Comment by bryanmacfarlane** [comment]
From root of exising agent, npm update vso-task-lib
Or deploy new agent

**Comment by Garyljackson** [comment]
Great, thanks Bryan.

**Comment by bryanmacfarlane** [comment]
Agent 0.2.23 and above pull the fix from vso-task-lib 0.4.4 and above
